airflow/providers/amazon/aws/auth_manager/aws_auth_manager.py

```diff
@@ -0,0 +1,145 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import warnings
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+from flask import session, url_for
+
+from airflow.exceptions import AirflowException, AirflowOptionalProviderFeatureException
+from airflow.providers.amazon.aws.auth_manager.security_manager.aws_security_manager_override import (
+    AwsSecurityManagerOverride,
+)
+
+try:
+    from airflow.auth.managers.base_auth_manager import BaseAuthManager, ResourceMethod
+except ImportError:
+    raise AirflowOptionalProviderFeatureException(
+        "Failed to import BaseUser. This feature is only available in Airflow versions >= 2.8.0"
+    )
+
+if TYPE_CHECKING:
+    from airflow.auth.managers.models.base_user import BaseUser
+    from airflow.auth.managers.models.resource_details import (
+        AccessView,
+        ConfigurationDetails,
+        ConnectionDetails,
+        DagAccessEntity,
+        DagDetails,
+        DatasetDetails,
+        PoolDetails,
+        VariableDetails,
+    )
+    from airflow.providers.amazon.aws.auth_manager.user import AwsAuthManagerUser
+    from airflow.www.extensions.init_appbuilder import AirflowAppBuilder
+
+
+class AwsAuthManager(BaseAuthManager):
+    """
+    AWS auth manager.
+
+    Leverages AWS services such as Amazon Identity Center and Amazon Verified Permissions to perform
+    authentication and authorization in Airflow.
+
+    :param appbuilder: the flask app builder
+    """
+
+    def __init__(self, appbuilder: AirflowAppBuilder) -> None:
+        super().__init__(appbuilder)
+        warnings.warn(
```

Comment:
Maybe throw a `NotImplementedError`?

Reply:
The only reason why I did not do that is that force me to have constantly local changes to test the auth manager. But I guess it makes more sense, let me do that

Reply:
A flag in config could be nice. But then is it a back compat issue to remove it in the future?

I think probably just dealing with the slight annoyance of commenting out that exception is the easiest thing to do.

Reply:
> But then is it a back compat issue to remove it in the future?

Setting a config that does not exist raise an error? I dont think so no? If not, there is no back compat issue :)

Reply:
This config will be to enable/disable this feature right? If people begin to make use of that to disable the feature, then it is removed and they can no longer control the feature anymore I think that might be back compat? But I do agree it's a stretch, more of a technicality. I'd be fine with it.

Reply:
It is disabled by default, so I guess users wont set this config to the default value. When ready, I'll remove this config flag, hence if some mysterious users still had set this config, it will be ignored